sleap_nn/data/pipelines.py

```diff
@@ -0,0 +1,89 @@
+"""This module defines high level pipeline configurations from providers/transformers.
+
+This allows for convenient ways to configure individual variants of common pipelines, as
+well as to define training vs inference versions based on the same configurations.
+"""
+import torch
+from omegaconf.dictconfig import DictConfig
+from torch.utils.data.datapipes.datapipe import IterDataPipe
+
+from sleap_nn.data.augmentation import KorniaAugmenter
+from sleap_nn.data.confidence_maps import ConfidenceMapGenerator
+from sleap_nn.data.instance_centroids import InstanceCentroidFinder
+from sleap_nn.data.instance_cropping import InstanceCropper
+from sleap_nn.data.normalization import Normalizer
+from sleap_nn.data.providers import LabelsReader
+
+
+class SleapDataset(IterDataPipe):
+    """Returns image and corresponding heatmap for the DataLoader.
+
+    This class is to return the image and its corresponding confidence map
+    to load the dataset with the DataLoader class
+
+    Attributes:
+    source_dp: The previous `DataPipe` with samples that contain an `instances` key.
+    """
+
+    def __init__(self, source_dp: IterDataPipe):
+        """Initialize SleapDataset with the source `DataPipe."""
+        self.dp = source_dp
+
+    def __iter__(self):
+        """Return a tuple with the cropped image and the heatmap."""
+        for example in self.dp:
+            if len(example["instance_image"].shape) == 4:
+                example["instance_image"] = example["instance_image"].squeeze(dim=0)
+            yield example["instance_image"], example["confidence_maps"]
```

Comment:
There will be other return signatures and I would like us to keep it general and just return the dictionary.

This should be a different block that either unpacks specified keys into a tuple, or just return a dictionary with only the specified keys so that we can drop the larger keys that aren't used (e.g., `"image"`).

It might look something like this: https://github.com/talmolab/sleap/blob/e4fca4f96b711648b13f4e572995e77d0a718ead/sleap/nn/data/general.py#L50-L77

Which you could add to this pipeline to specify the keys to keep in the output dictionary. I prefer we do that over returning a tuple since it keeps the tensors organized and easy to pull out in the `train_step` method of the trainer downstream.